### ISSUE TITLE
ACR not attaching to AKS despite being the owner of subscription.

### ISSUE THREAD

**Issue by fahimhoque** [Azure/AKS#3530]
I am trying to attach ACR to AKS cluster but all I am getting is 

`%Could not create a role assignment for ACR. Are you an Owner on this subscription?`

Now here are the screenshots of the process I have done so far:

1.
<img width="935" alt="Screenshot 2023-03-12 at 5 45 21 PM" src="https://user-images.githubusercontent.com/50498916/224542811-15f07930-adde-4492-938d-7838faa75d00.png">

2.
<img width="935" alt="Screenshot 2023-03-12 at 5 45 54 PM" src="https://user-images.githubusercontent.com/50498916/224542833-d08ad7b8-a9c7-49c1-9785-13ba9da64091.png">

3.
<img width="935" alt="Screenshot 2023-03-12 at 5 46 09 PM" src="https://user-images.githubusercontent.com/50498916/224542845-c227688c-a0e1-4284-b025-9c0bced7ba26.png">

As you can see the final command: `az role assignment list --all --output table` shows be as the owner of the subscription. Running `az --version` gives:

```
azure-cli                         2.46.0

core                              2.46.0
telemetry                          1.0.8

Dependencies:
msal                              1.20.0
azure-mgmt-resource             21.1.0b1

Python location '/opt/homebrew/Cellar/azure-cli/2.46.0/libexec/bin/python'
Extensions directory '/Users/fahimhoque/.azure/cliextensions'

Python (Darwin) 3.10.10 (main, Feb 16 2023, 02:49:39) [Clang 14.0.0 (clang-1400.0.29.202)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by boliu83** [comment]
can you check if the "<cluster_name>-agentpool" identity has already got the AcrPull role assignment on this registry?  because that can cause the same error message too.

**Comment by jaliyaudagedara** [comment]
I am facing the same issue. In my case, I am trying to attach an ACR in a different subscription to an existing AKS.

Similar issue #1517

**Comment by jaliyaudagedara** [comment]
Ran the `aks update --attach-acr` with `--debug` flag.

I am seeing PUT calls to assign the role, all responding with ` (RoleAssignmentExists) The role assignment already exists.`. And ultimately ended up with this.
```
cli.azure.cli.command_modules.acs._roleassignments: (RoleAssignmentExists) The role assignment already exists.
Code: RoleAssignmentExists
Message: The role assignment already exists.
cli.azure.cli.core.azclierror: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acs/custom.py", line 603, in aks_update
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acs/managed_cluster_decorator.py", line 6513, in update_mc_profile_default
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acs/managed_cluster_decorator.py", line 5944, in process_attach_detach_acr
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acs/_roleassignments.py", line 332, in ensure_aks_acr
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acs/_roleassignments.py", line 316, in ensure_aks_acr_role_assignment
azure.cli.core.azclierror.AzCLIError: Could not create a role assignment for ACR. Are you an Owner on this subscription?

cli.azure.cli.core.azclierror: Could not create a role assignment for ACR. Are you an Owner on this subscription?
az_command_data_logger: Could not create a role assignment for ACR. Are you an Owner on this subscription?
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x045631D8>]
az_command_data_logger: exit code: 1
cli.__main__: Command ran in 147.360 seconds (init: 0.453, invoke: 146.907)
telemetry.main: Begin splitting cli events and extra events, total events: 1
telemetry.client: Accumulated 0 events. Flush the clients.
telemetry.main: Finish splitting cli events and extra events, cli events: 1
telemetry.save: Save telemetry record of length 3344 in cache
```

**Comment by mottimalka** [comment]
Foe me the issue was the Role Assignment already exits, the error message from the output command not show this, after run the attach command with `--debug` mode i able to see the error that this already exists

**Comment by Misterjohnson87** [comment]
This is an issue with Azure CLI 2.46.0

I was able to run the `aks update --attach-acr` fine when using version 2.45.0, as soon as I upgraded to 2.46.0 I got the error - `azure.cli.core.azclierror.AzCLIError: Could not create a role assignment for ACR. Are you an Owner on this subscription?`

**Comment by jaliyaudagedara** [comment]
Checked this with Azure CLI **`2.47.0`**.

Seems the issue is fixed.

![image](https://user-images.githubusercontent.com/5653381/230750041-97cedd6b-4e5b-4edb-ac29-cb5838f99ce4.png)

**Comment by davidbarrowsatasos** [comment]
I can also say this failed for me using 2.46.0; when I upgraded to 2.48.0 using "az upgrade", then re-ran the "az aks update --attach-acr" command, it worked.

**Comment by sabbour** [comment]
Looks like this was fixed in Azure CLI 2.47.0
